src/lib/getEventType.ts

```diff
@@ -0,0 +1,192 @@
+import { ParsedQueryString } from "./models";
+
+type BrokenEvent = "Broken Event";
+type PingType =
+  | "HeartBeat"
+  | "Deanonymization"
+  | "Page performance metric"
+  | "Custom"
+  | BrokenEvent;
+type EventType =
+  | "Goal Conversion"
+  | "Ping"
+  | "Download"
+  | "Outlink"
+  | "Consent form impression"
+  | "Consent form click"
+  | "Consent decision"
+  | "SharePoint"
+  | "Custom event"
+  | "Content interaction"
+  | "Content impression"
+  | "Cart update"
+  | "Product detail view"
+  | "Add to cart"
+  | "Remove from cart"
+  | "Order completed"
+  | "Internal search"
+  | "Page view"
+  | PingType
+  | BrokenEvent;
+/**
+ * https://help.piwik.pro/support/questions/what-are-events-and-how-are-they-detected/
+ */
+export function getEventType(eventParams: ParsedQueryString[]): EventType {
+  if (isGoalConversion(eventParams)) return "Goal Conversion";
+
+  if (isPing(eventParams)) return getPingType(eventParams);
+
+  if (isDownload(eventParams)) return "Download";
+
+  if (isOutlink(eventParams)) return "Outlink";
+
+  if (isConsentFormImpression(eventParams)) return "Consent form impression";
+
+  if (isConsentFormClick(eventParams)) return "Consent form click";
+
+  if (isConsentDecision(eventParams)) return "Consent decision";
+
+  if (isSharePoint(eventParams)) return "SharePoint";
+
+  if (isCustomEvent(eventParams)) return "Custom event";
+
+  if (isContentInteraction(eventParams)) return "Content interaction";
+
+  if (isContentImpression(eventParams)) return "Content impression";
+
+  if (isCartUpdate(eventParams)) return "Cart update";
+
+  if (isProductDetailView(eventParams)) return "Product detail view";
+
+  if (isAddToCart(eventParams)) return "Add to cart";
+
+  if (isRemoveFromCart(eventParams)) return "Remove from cart";
+
+  if (isOrderCompleted(eventParams)) return "Order completed";
+
+  if (isInternalSearch(eventParams)) return "Internal search";
+
+  return "Page view";
+}
+
+function isGoalConversion(eventParams: ParsedQueryString[]) {
+  const idgoal = eventParams.find((p) => p.name === "idgoal");
+  return idgoal?.value && idgoal.value !== "0";
```

Comment:
So I can't really wrap my head around why `idgoal?.value !== "0";` isn't sufficient enough. If it isn't enough here, isn't it also for the case for other checks like `isConsentFormImpression` or `isOrderCompleted` that follow the same convention?

Reply:
If I remember correctly, I tested a simple page view request and it was categorized as a goal conversion. Since the only thing this function checked in the end was whether the value is not equal to 0, it also returned true when `undefined` was returned by `idgoal?.value`.

Reply:
oh, i got it, this is the only place where we check if value is **unequal**. Thanks for catching that!